spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/DistriParameterSynchronizer.scala

```diff
@@ -0,0 +1,429 @@
+   * @param name  identifier for parameter
+   * @param globalSize total size of parameter
+   * @param priority priority for this parameter
+   */
+  def init(name: String, globalSize: Int, priority: Int = 1): Unit
+
+  /**
+   * put parameter to global
+   * @param name identifier for parameter
+   * @param parameter  parameter to put
+   */
+  def put(name: String, parameter: Tensor[T]): Unit
+
+  /**
+   * get parameter for specific identifier
+   * @param name  identifier for parameter
+   * @return  parameter
+   */
+  def get(name: String): Tensor[T]
+
+  /**
+   * clear the synchronizer
+   */
+  def clear(): Unit
+}
+
+class BlockManagerParameterSynchronizer[T: ClassTag](partitionID: Int,
+                                                     totalPartition: Int)
+                                                    (implicit ev: TensorNumeric[T])
+  extends DistriParameterSynchronizer[T] {
+
+  import com.intel.analytics.bigdl.utils.BlockManagerParameterSynchronizer.logger
+
+  @volatile private var shutdown = false
+
+  private val syncResults: mutable.HashMap[String, FutureTask[Tensor[T]]]
+    = new mutable.HashMap[String, FutureTask[Tensor[T]]]()
+
+  private val taskSize: Int = System.getProperty("bigdl.ParameterSynchronier." +
+    "asyncTaskSize", "100").toInt
+
+  private val clearPoolSize: Int = System.getProperty("bigdl.ParameterSynchronier." +
+    "clearPoolSize", "1").toInt
+
+  private val workerPoolSize: Int = System.getProperty("bigdl.ParameterSynchronier" +
+    ".syncPoolSize", "4").toInt
+
+  private val syncPoolSize: Int = Math.max(System.getProperty("bigdl.ParameterSynchronier" +
+    ".computePoolSize",
+    (Runtime.getRuntime().availableProcessors() / 2).toString).toInt, 2)
+
+  private val fetchCompletionPoolSize: Int = System.getProperty("bigdl.ParameterSynchronier" +
+    ".fetchCompletionPoolSize", "1").toInt
+
+  private val asyncTaskWaitingQueue : PriorityBlockingQueue[AsyncFutureTask] =
+    new PriorityBlockingQueue[AsyncFutureTask](taskSize)
+
+  private val blockFetchRequestQueue: PriorityBlockingQueue[BlockFetchRequest] =
+    new PriorityBlockingQueue[BlockFetchRequest](taskSize)
+
+  // thread pool to remove expired blocks
+  private lazy val clearPool: ExecutorService =
+    Executors.newFixedThreadPool(clearPoolSize, new ThreadFactory {
+      override def newThread(r: Runnable): Thread = {
+        val t = Executors.defaultThreadFactory().newThread(r)
+        t.setDaemon(true)
+        t
+      }
+    })
+
+  // main thread pool to do put-get-aggregate
+  private val workerPool: ExecutorService =
+    Executors.newFixedThreadPool(workerPoolSize, new ThreadFactory {
+    override def newThread(r: Runnable): Thread = {
+      val t = Executors.defaultThreadFactory().newThread(r)
+      t.setDaemon(true)
+      t
+    }
+  })
+
+  // long running thread to fetch the request
+  workerPool.submit(new Runnable {
+    override def run(): Unit = {
+      val asyncFutureTask = asyncTaskWaitingQueue.take
```

Comment:
long running?

Reply:
it's a bug, thanks for pointing out, just fixed it